CMSgov/easi-app
# [EASI-1394] CEDAR core client

# EASI-1394



## Changes proposed in this pull request

- Introduces a new `pkg/cedar/core` folder for the CEDAR Core API code
  - Code generated by `go-swagger` lives in `pkg/cedar/core/gen`
  - The translated client (that should be used by the rest of the code base) is in `pkg/cedar/core/client.go` 



## Reviewer Notes



## Setup & How to test

1. If you don't already have one, create a `.envrc.local` file at the root of the project, and add the following to it:
```
export CEDAR_API_URL="webmethods-apigw.cedarimpl.cms.gov"
export CEDAR_API_KEY=THE_IMPL_KEY
```
2. Run `direnv allow` to pull in the new env variables
3. Run `scripts/dev up` to start EASi locally
4. In `/pkg/cedar/core/client.go`, uncomment the section labeled `// Uncomment the following if you're having trouble with TLS locally` (there's probably a better way to handle this, but the other clients had something similar)
5. Run `curl http://localhost:8080/system-info-DEBUG`
6. Check that the output looks correct!


- Check that all code is adequately covered by tests
- Make comments, even if it's minor!



## PR Author Review Checklist

functions, or e2e tests as necessary
the [rotor menu](https://support.apple.com/guide/voiceover/with-the-voiceover-rotor-mchlp2719/mac)
- For any new migrations/schema changes:

## Screenshots
